operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/Controller.java

```diff
@@ -179,26 +187,43 @@ public String successTypeName(UpdateControl<P> result) {
           @Override
           public UpdateControl<P> execute() throws Exception {
             initContextIfNeeded(resource, context);
-            dependents.forEach(dependent -> dependent.reconcile(resource, context));
+            final var result = dependents.entrySet().stream()
+                .map(entry -> {
+                  final var dependent = entry.getValue();
+                  final var name = entry.getKey();
+                  ReconcileResult reconcileResult;
+                  try {
+                    reconcileResult = dependent.reconcile(resource, context);
+                  } catch (Exception e) {
+                    reconcileResult = ReconcileResult.error(resource, e);
+                  }
+                  context.managedDependentResourceContext().setReconcileResult(name,
+                      reconcileResult);
+                  return reconcileResult;
+                });
+
+            log.info("Dependents reconciliation:\n{}",
+                result.map(r -> "\t" + r.toString()).collect(Collectors.joining("\n")));
             return reconciler.reconcile(resource, context);
           }
         });
   }
 
-  @Override
-  public List<EventSource> prepareEventSources(EventSourceContext<P> context) {
-    List<EventSource> sources = new LinkedList<>();
-    dependents.stream()
-        .filter(dependentResource -> dependentResource instanceof EventSourceProvider)
-        .map(EventSourceProvider.class::cast)
-        .map(provider -> provider.initEventSource(context))
-        .forEach(sources::add);
+  public void initAndRegisterEventSources(EventSourceContext<P> context) {
+    dependents.entrySet().stream()
+        .filter(drEntry -> drEntry.getValue() instanceof EventSourceProvider)
+        .forEach(drEntry -> {
+          final var provider = (EventSourceProvider) drEntry.getValue();
+          final var source = provider.initEventSource(context);
```

Comment:
Could we use here the name of the dependent resource to register the event source?

Reply:
That would solve the same naming problem.

Reply:
This is exactly what's done on the next line or did I misunderstood your comment?

Reply:
Sorry, did not see that, my fault. Great!